plugin/plugin.go

```diff
@@ -162,6 +137,108 @@ func (e *Ratelimit) Type() rules.ActionType {
 	return rules.ActionTypeNondisruptive
 }
 
+func (e *Ratelimit) parseConfig(config string) error {
+	// acceptable keys
+	var err error
+
+	tokens := strings.Split(config, "&")
+
+	requiredValues := map[string]bool{
+		"zone":   false,
+		"events": false,
+		"window": false,
+	}
+
+	for _, token := range tokens {
+		pair := strings.Split(token, "=")
+		if len(pair) != 2 {
+			return fmt.Errorf("invalid usage of = for %v", pair)
+		}
+
+		key := pair[0]
+		value := pair[1]
+
+		switch key {
+		case "zone":
+			if e.ZoneMacro, err = macro.NewMacro(value); err != nil {
+				return fmt.Errorf("invalid macro name")
```

Comment:
Use `errors.New` as you don't need to do interpolation

Reply:
done